### PR TITLE
Build lib/addons with ts-up

### PR DESCRIPTION
**Pull request by abdlqader**
Issue: #18732

## What I did
Changed addons in lib to be compiled by the bundler ( ts-up )
## How to test

- [ YES ] Is this testable with Jest or Chromatic screenshots? Testable by the original tests
- [ NO ] Does this need a new example in the kitchen sink apps?
- [ NO ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: code/lib/addons/package.json [hunk 1]
```diff
@@ -31,7 +39,17 @@
...
+    "prepare": "../../../scripts/prepare/bundle.ts"
+  },
+  "bundler": {
+    "entries": [
+      "./src/index.ts",
+      "./src/hooks.ts",
+      "./src/main.ts",
+      "./src/make-decorator.ts",
+      "./src/storybook-channel-mock.ts",
+      "./src/types.ts"
```
Comment (ndelangen) [inline code/lib/addons/package.json:51]: I have the feeling, that only `./src/index.ts` is needed here.

Comment (abdlqader) [reply code/lib/addons/package.json]: Solved.

**Comment by IanVS** [comment]
Shoot I rebased this and I thought it would keep the original author of the commits.  Let me try to fix this...

**Comment by IanVS** [comment]
@ndelangen I got this passing by adding `/// <reference types="webpack-env" />` to the addons that are using `module.hot.decline()`, since that's a webpack feature.  

Which makes me wonder, do we need these to also call `import.meta.hot.decline()` for the vite builder?

**Comment by ndelangen** [comment]
Yes that would make sense for any files that are referenced from stories

**Review by ndelangen — APPROVED** [review]

**Pull request merged**